tests/v2_tests/test_create_index.py

```diff
@@ -424,4 +433,116 @@ def test_create_structured_index_with_languagebind(self):
+                with self.assertRaises(ValueError) as e:
+                    self.client.create_index(
+                        index_name=self.index_name,
+                        **{key: parameters_pool[key]},
+                        settings_dict={"type": "unstructured"}
+                    )
+                self.assertIn(
+                    f"'settings_dict' cannot be specified with other index creation parameters.",
+                    str(e.exception)
+                )
+
+    def test_create_index_SettingsDictCanNotBeSpecificWithOtherParametersCloud(self):
+        """Test that settings_dict cannot be specified with other index creation
+        parameters in cloud create_index call."""
+        parameters_pool = {
+            "inference_type": "marqo.CPU.large",
+            "storage_class": "marqo.basic",
+            "number_of_replicas": 1,
+            "number_of_shards": 1,
+            "number_of_inferences": 1
+        }
+
+        test_cloud_client = Client("https://api.marqo.ai", api_key="test")
+        for key in parameters_pool.keys():
+            with self.subTest(f"key={key}"):
+                with self.assertRaises(ValueError) as e:
+                    test_cloud_client.create_index(
+                        index_name=self.index_name,
+                        **{key: parameters_pool[key]},
+                        settings_dict={"type": "unstructured"}
+                    )
+                self.assertIn(
+                    f"'settings_dict' cannot be specified with other index creation parameters.",
+                    str(e.exception)
+                )
+
+    def test_CloudIndexSettingsGenerateRequestBody(self):
+        """Test that CloudIndexSettings generate_request_body method returns the correct request body."""
+        expected_request_body = {
+            "type": "unstructured",
+        }
+        cloud_index_settings = CloudIndexSettings(
+            type="unstructured"
+        )
+        self.assertEqual(expected_request_body, cloud_index_settings.generate_request_body())
+
+@mark.fixed
+@mark.cloud_only_tests
+class TestCloudCreateIndex(MarqoTestCase):
+
+    TEST_ATTRIBUTES_WITH_DEFAULTS = {
+        "inferenceType": "CPU.SMALL",
+        "numberOfShards": 1,
+        "numberOfReplicas": 0,
+        "numberOfInferences": 1,
+        "storageClass": "BASIC"
+    }
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.get_index_response: List[Dict] = requests.get(
+            f"{cls.authorized_url}/api/v2/indexes",
+            headers={"x-api-key": os.environ.get("MARQO_API_KEY", None)}
+        ).json()["results"]
+
+    def test_cloud_index_attributes(self):
+        test_indexes = list(index_name_to_settings_mappings.keys())
+        for test_index in test_indexes:
+            index_name = test_index + "_" + os.environ.get("MQ_TEST_RUN_IDENTIFIER", "")
+            with self.subTest(f"Index name: {index_name}"):
```

Comment:
Why do we need to test each Cloud index? Will it need to create a cloud index for each sub test? How long would this test take?

Reply:
No, this test does not create any indexes. It only tests on the created indexes.